sycl/source/detail/graph_impl.cpp

```diff
@@ -1551,56 +1554,46 @@ bool modifiable_command_graph::begin_recording(queue &RecordingQueue) {
     QueueImpl->setCommandGraph(impl);
     graph_impl::WriteLock Lock(impl->MMutex);
     impl->addQueue(QueueImpl);
-    return true;
   }
   if (QueueImpl->getCommandGraph() != impl) {
     throw sycl::exception(sycl::make_error_code(errc::invalid),
                           "begin_recording called for a queue which is already "
                           "recording to a different graph.");
   }
-  // Queue was already recording to this graph.
-  return false;
 }
 
-bool modifiable_command_graph::begin_recording(
-    const std::vector<queue> &RecordingQueues) {
-  bool QueueStateChanged = false;
+void modifiable_command_graph::begin_recording(
+    const std::vector<queue> &RecordingQueues,
+    const sycl::property_list &PropList) {
   for (queue Queue : RecordingQueues) {
```

Comment:
```suggestion
  for (queue &Queue : RecordingQueues) {
```
Not part of this PR technically but since we're here anyway this should probably be a reference.

Reply:
As discussed in the meeting, this change cannot be done because it would result in a compiler error due to discarding the const value from the queue elements.